packages/nimble-components/src/table/specs/README.md

```diff
@@ -132,13 +132,17 @@ The various APIs/features of the `nimble-table` will be split up amongst several
     -   Define how the action menu gets associated with a particular column
     -   Define the table-level(column-level?) API(s) for applying an action menu to the table (slot, properties, etc...)
 
+_Attributes_
+
+-   `id-field-name` - An optional string attribute that specifies the field name within a row's record to use as a row's ID. If the attribute is not defined or a field with the specified name does not exist in a row's record, a default ID will be created. The default ID is the index of the row's record within the table's `data` property. The value associated with the field name must be unique across all records in the table's data.
```

Comment:
Is it too early to define that the [CellState](https://github.com/ni/nimble/blob/bec417c55304b1cb23de95af2a8e8e35c6658019/packages/nimble-components/src/table/specs/table-columns-hld.md#tablecellstate-interface) will include the id / calculatedId / resolvedId?

Seems like it would be useful to also communicate with the cell what the id mode is. So if a button in a cell template fires an event it can include `detail: {resolvedId: 7, idMode: 'index'}` vs `detail: {resolvedId: 'user-defined-id-8', idMode: 'field'}` or something.

Reply:
I've updated `table-columns-hld.md` to include `rowId: string` in the `TableCellState` interface. I don't think we'll have a concept of ID modes though.